### PR TITLE
Migrate Refactoring for Generating printOn: method

### PR DESCRIPTION
**Pull request by hernanmd**
# Description

This PR migrates the refactoring browser to generate the #printOn: method for the new driver interface. The new interface provides a more streamlined and extensible approach for generating these methods, making it easier to maintain and enhance the functionality in the future.

# Motivation

The existing implementation of the refactoring browser for generating the #printOn: method was tightly coupled with the user interface (UI) code.

# Implementation.

The implementation of this PR involves the following steps:

- Create a new driver interface: A new driver interface, `ReGeneratePrintOnDriver,` has been introduced. This interface defines the contract for generating the #printOn: method and encapsulates the core logic for this functionality.
- Implement the driver: The driver uses `ReGeneratePrintOnTransformation` for the underlying logic.
- UI interaction: The driver uses `SpSelectMultipleDialog` to select the variables in the generated #printOn: method.
- Created tests: Tests related to the generation of the #printOn: method has been provided in `ReGeneratePrintOnDriverTest`
- Additionally, the PR includes a superclass `ReGenerateAbstractTransformation` to group an instance variable and common methods to `ReGenerateEqualHashTransformation` and `ReGeneratePrintOnTransformation` 
- Similarly, the `ReGenerateMethodDriver` abstracts behavior common to `ReGenerateEqualAndHashDriver` and ` ReGeneratePrintOnDriver`.

### CODE REVIEW

File: src/Refactoring-UI/ReGenerateEqualAndHashDriver.class.st [hunk 1]
```diff
@@ -6,93 +6,20 @@ It uses the `RBGenerateEqualHashTransformation`.
...
 	#package : 'Refactoring-UI',
 	#tag : 'Drivers'
 }
 
-{ #category : 'factory method' }
+{ #category : 'displaying' }
 ReGenerateEqualAndHashDriver class >> refactoringClass [
 
 	^ ReGenerateEqualHashTransformation
 ]
```
Comment (balsa-sarenac) [inline src/Refactoring-UI/ReGenerateEqualAndHashDriver.class.st:19]: What is the reason for storing class name on class side?

File: src/SystemCommands-ClassCommands/SycCmGenerateCommand.class.st [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+		 targetClass: targetClass name;
+		 scopes: refactoringScopes) runRefactoring
+]
+
+{ #category : 'testing' }
+SycCmGenerateCommand >> isApplicable [
+	"Answer <true> if the receiver has any variables"
+
+	^ false
+]
```
Comment (balsa-sarenac) [inline src/SystemCommands-ClassCommands/SycCmGenerateCommand.class.st:32]: Shouldn't this be something like `targetClass instnaceVariables isNotEmpty`?

**Review by balsa-sarenac — COMMENTED** [review]
Nice work! Left two minor comments

File: src/Refactoring-UI/ReGenerateEqualAndHashDriver.class.st (hunk 1 above)
Comment (hernanmd) [reply src/Refactoring-UI/ReGenerateEqualAndHashDriver.class.st:19]: This is to avoid `MyClass new` to get `refactoringClass`.

Comment (hernanmd) [reply src/Refactoring-UI/ReGenerateEqualAndHashDriver.class.st:19]: Can I set this as Resolved so we can move on?

Comment (balsa-sarenac) [reply src/Refactoring-UI/ReGenerateEqualAndHashDriver.class.st:19]: yes

**Review by balsa-sarenac — APPROVED** [review]

**Pull request merged**